source/backend/api/Services/FormDocumentService.cs

```diff
@@ -0,0 +1,136 @@
+using Pims.Dal.Repositories;
+using Pims.Dal.Security;
+
+namespace Pims.Api.Services
+{
+    public class FormDocumentService : BaseService, IFormDocumentService
+    {
+        private readonly ILogger _logger;
+        private readonly IMapper _mapper;
+        private readonly IFormTypeRepository _formTypeRepository;
+        private readonly IDocumentService _documentService;
+        private readonly IDocumentRepository _documentRepository;
+        private readonly IAcquisitionFileFormRepository _acquisitionFileFormRepository;
+
+        public FormDocumentService(
+            ClaimsPrincipal user,
+            ILogger<FormDocumentService> logger,
+            IMapper mapper,
+            IFormTypeRepository formTypeRepository,
+            IDocumentService documentService,
+            IDocumentRepository documentRepository,
+            IAcquisitionFileFormRepository acquisitionFileFormRepository)
+            : base(user, logger)
+        {
+            _logger = logger;
+            _mapper = mapper;
+            _formTypeRepository = formTypeRepository;
+            _documentService = documentService;
+            _documentRepository = documentRepository;
+            _acquisitionFileFormRepository = acquisitionFileFormRepository;
+        }
+
+        public IList<PimsFormType> GetAllFormDocumentTypes()
+        {
+            _logger.LogInformation("Getting all form document types");
+            this.User.ThrowIfNotAuthorized(Permissions.FormView);
+
+            var formTemplates = _formTypeRepository.GetAllFormTypes();
+            return formTemplates;
+        }
+
+        public IList<PimsFormType> GetFormDocumentTypes(string formTypeCode)
+        {
+            this.Logger.LogInformation("Getting document types for code {formTypeCode}", formTypeCode);
+            this.User.ThrowIfNotAuthorized(Permissions.FormView);
+
+            return new List<PimsFormType>() { _formTypeRepository.GetByFormTypeCode(formTypeCode) };
+        }
+
+        public async Task<DocumentUploadRelationshipResponse> UploadFormDocumentTemplateAsync(string formTypeCode, DocumentUploadRequest uploadRequest)
+        {
+            this.Logger.LogInformation("Uploading template for document form type");
+            this.User.ThrowIfNotAuthorized(Permissions.DocumentAdmin);
+
+            var currentFormType = _formTypeRepository.GetByFormTypeCode(formTypeCode);
+            if (currentFormType.DocumentId != null)
+            {
+                var result = await DeleteFormDocumentTemplateAsync(currentFormType);
+                if (result.Status != ExternalResultStatus.Success)
+                {
+                    throw new InvalidOperationException("Could not remove existing template");
+                }
+                else
+                {
+                    // Retrieve the form type again to get the latest after deletion.
+                    currentFormType = _formTypeRepository.GetByFormTypeCode(formTypeCode);
```

Comment:
Hmm, why do we need this? does the DeleteFormDocumentTemplateAsync call make a change to the database as well as mayan?

Reply:
It follows the usual process for deleting documents; If there is only one, then it deletes both the mayan stored one and the pims reference to it,